staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -1452,14 +1462,19 @@ func TestGCECondition(t *testing.T) {
 	cloudNodeController := &CloudNodeController{
 		kubeClient:                clientset,
 		nodeInformer:              factory.Core().V1().Nodes(),
+		nodesLister:               factory.Core().V1().Nodes().Lister(),
 		cloud:                     fakeCloud,
 		recorder:                  eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"}),
 		nodeStatusUpdateFrequency: 1 * time.Second,
 	}
+
+	factory.Start(nil)
+	factory.WaitForCacheSync(nil)
+
```

Comment:
@andrewsykim I got a wired thing, even if I set up the lister correctly, I still get 
```
--- FAIL: TestGCEConditionV2 (0.10s)
    node_controller_test.go:1403:
        	Error Trace:	node_controller_test.go:1403
        	Error:      	Should be true
        	Test:       	TestGCEConditionV2
        	Messages:   	Network Route Condition for GCE not added by external cloud initializer
I0915 13:56:13.190000   55392 node_controller.go:438] Initializing node node0 with cloud provider
I0915 13:56:13.190204   55392 node_controller.go:472] Successfully initialized node node0 with cloud provider
I0915 13:56:13.190262   55392 event.go:282] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"node0", UID:"", APIVersion:"v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'Synced' Node synced successfully
--- FAIL: TestGCECondition (0.10s)
    node_controller_test.go:1491:
        	Error Trace:	node_controller_test.go:1491
        	Error:      	Should be true
        	Test:       	TestGCECondition
        	Messages:   	Network Route Condition for GCE not added by external cloud initializer
FAIL
FAIL	k8s.io/cloud-provider/controllers/node	1.959s
FAIL
```
But the logs look fine
And when I trigger the test directly from my IDE, I got the tests running correctly, but running "go test ./..." commands will show the failing tests with above message

Reply:
and https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/94736/pull-kubernetes-bazel-test/1305973363159601155 logs the error 🤔 But not sure where goes wrong